### PR TITLE
autotiling: init at 1.1

### PR DESCRIPTION
**Pull request by Artturin**
autotiling for i3 & sway
###### Motivation for this change
I use it and i think it will be useful for other people too

closes: #91244
###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/autotiling/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  pname = "autotiling";
+  version = "1.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0hwvy9bxwv9fakqqiyrkmpckxgm0z85c240p84ibdhja9sm086v0";
+  };
+
+  propagatedBuildInputs = [ i3ipc importlib-metadata ];
+  doCheck = false;
```
Comment (ryantm) [inline pkgs/misc/autotiling/default.nix:13]: Why aren't the checks working?

**Review by ryantm — CHANGES_REQUESTED** [review]

File: pkgs/misc/autotiling/default.nix (hunk 1 above)
Comment (Artturin) [reply pkgs/misc/autotiling/default.nix:13]: wrapping '/nix/store/pmsr2q6r0cw67rhvnhh61azw61aky8fg-autotiling-1.1/bin/autotiling'...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
running install tests
no Makefile or custom buildPhase, doing nothing
pythonCatchConflictsPhase
pythonRemoveBinBytecodePhase
pythonImportsCheckPhase
Executing pythonImportsCheckPhase
setuptoolsCheckPhase
Executing setuptoolsCheckPhase
running test
WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent of test runner are encouraged to use tox.
running egg_info
writing autotiling.egg-info/PKG-INFO
writing dependency_links to autotiling.egg-info/dependency_links.txt
writing entry points to autotiling.egg-info/entry_points.txt
writing requirements to autotiling.egg-info/requires.txt
writing top-level names to autotiling.egg-info/top_level.txt
reading manifest file 'autotiling.egg-info/SOURCES.txt'
writing manifest file 'autotiling.egg-info/SOURCES.txt'
running build_ext
usage: nix_run_setup
       [-h]
       [--debug]
nix_run_setup: error: unrecognized arguments: test
builder for '/nix/store/g4l6jjhc55lpk176zbkk8y5q1ks87y7j-autotiling-1.1.drv' failed with exit code 2

Comment (Artturin) [reply pkgs/misc/autotiling/default.nix:13]: found a solution just now https://github.com/NixOS/nixpkgs/pull/64529#discussion_r326165509

checkPhase = ''
    ${python.interpreter} test.py
'';

File: pkgs/misc/autotiling/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  meta = with lib; {
+    homepage = "https://github.com/nwg-piotr/autotiling";
+    description = "Script for sway and i3 to automatically switch the horizontal / vertical window split orientation";
+    license = licenses.gpl3;
```
Comment (sikmir) [inline pkgs/misc/autotiling/default.nix:18]:
```suggestion
    license = licenses.gpl3Plus;
```
See https://github.com/nwg-piotr/autotiling/blob/126c07da532fd58df0af86f9f1741e29d6bf2db9/setup.cfg#L13

Comment (Artturin) [reply pkgs/misc/autotiling/default.nix:18]: okay, i will do it in the same commit as the above request if i need to change something in the expression

Comment (Artturin) [reply pkgs/misc/autotiling/default.nix]: done

File: pkgs/misc/autotiling/default.nix (hunk 1 above)
Comment (ryantm) [reply pkgs/misc/autotiling/default.nix:13]: I looked through the project and there appear to be no tests.

**Comment by ryantm** [comment]
@Artturin Grats on your first PR!

**Comment by Artturin** [comment]
@ryantm thank you very much!

File: pkgs/misc/autotiling/default.nix (hunk 1 above)
Comment (Artturin) [reply pkgs/misc/autotiling/default.nix:13]: okay, i thought that nix had its own tests. in the future i will mention it in the description if there are no tests.

**Pull request merged**